UI/MeterBrowseWnd.cpp

```diff
@@ -580,4 +585,360 @@ void ShipDamageBrowseWnd::UpdateEffectLabelsAndValues(GG::Y& top) {
+        damage_value->SetFont(ClientUI::GetBoldFont());
+        AttachChild(damage_value);
+        m_effect_labels_and_values.push_back(std::make_pair(damage_label, damage_value));
+
+        top += m_row_height;
+    } else {
+        // add labels for bay parts
+        for (std::multimap<std::string, int>::value_type& bay_part : bay_parts) {
+            const std::string& part_name = UserString(bay_part.first);
+            int fighter_total = bay_part.second;
+            m_bay_list->Insert(new ShipFightersBrowseRow(part_name, fighter_total));
+        }
+
+        // add a label for each hangar part, distributing the total fighters equally
+        int unaccounted_current_fighters = hangar_current_fighters;
+        int unaccounted_total_fighters = hangar_total_capacity;
+        const std::string& hangar_name = UserString(hangar_part.first);
+        for (int i = 0; i < hangar_part.second; ++i) {
+            int max_fighters = std::min(unaccounted_total_fighters, hangar_total_capacity / hangar_part.second);
+            // top heavy when current fighters do not distribute equally
+            int cur_fighters = std::min(unaccounted_current_fighters, std::min(max_fighters,
+                boost::math::iround(static_cast<float>(hangar_current_fighters) / hangar_part.second)));
+            unaccounted_current_fighters -= cur_fighters;
+            unaccounted_total_fighters -= max_fighters;
+            m_hangar_list->Insert(new ShipFightersBrowseRow(hangar_name, cur_fighters, max_fighters));
+        }
+        // validate that all fighters were accounted for
+        if (unaccounted_current_fighters != 0 || unaccounted_total_fighters != 0) {
+            DebugLogger() << "Display of number of fighters mismatched.  Current: "
+                          << unaccounted_current_fighters << "/" << hangar_current_fighters
+                          << " Total: " << unaccounted_total_fighters << "/" << hangar_total_capacity
+                          << "  For ship " << ship->ID() << " with design: \n" << design->Dump();
+        }
+
+        int max_rows = static_cast<int>(std::max(m_bay_list->NumRows(), m_hangar_list->NumRows()));
+        int list_vpad = max_rows > 0 ? 10 : 0;
+        const GG::Y LIST_HEIGHT = (m_row_height * max_rows) + list_vpad;
+
+        m_bay_list->MoveTo(GG::Pt(GG::X0, top));
+        m_bay_list->Resize(GG::Pt(ROW_WIDTH, LIST_HEIGHT));
+        m_hangar_list->MoveTo(GG::Pt(ROW_WIDTH + EDGE_PAD, top));
+        m_hangar_list->Resize(GG::Pt(ROW_WIDTH, LIST_HEIGHT));
+
+        top += LIST_HEIGHT;
+
+        // Damage summary labels
+        float total_dmg = FightersDamageForBout(bay_total_capacity, hangar_current_fighters, fighter_damage).second;
+        std::string detail_avg_text = boost::io::str(FlexibleFormat(UserString("SHIP_FIGHTERS_DAMAGE_AVERAGE"))
+            % ColourWrappedtext(DoubleToString(total_dmg / NUM_COMBAT_BOUTS, 3, false), DAMAGE_COLOR));
```

Comment:
indenting in this case is inconsistent throughout these additions. should perhaps be consistent with indendenting 4 speces or to the same position as FlexibleFormat